ros2topic/ros2topic/verb/pub.py

```diff
@@ -147,14 +153,19 @@ def publisher(
     pub = node.create_publisher(msg_module, topic_name, qos_profile)
 
     times_since_last_log = 0
+    total_wait_time = 0
     while pub.get_subscription_count() < wait_matching_subscriptions:
         # Print a message reporting we're waiting each 1s, check condition each 100ms.
         if not times_since_last_log:
             print(
                 f'Waiting for at least {wait_matching_subscriptions} matching subscription(s)...')
+        if max_wait_time is not None and max_wait_time <= total_wait_time:
+            return 'Timed out waiting for subscribers'       
         times_since_last_log = (times_since_last_log + 1) % 10
-        time.sleep(0.1)
-
+        WAIT_TIME = 0.1
```

Comment:
I am not familiar with how python interprets the code at runtime. but considering based C/C++, probably we can avoid store the same data to the variable? instead, we can declare this `WAIT_TIME` after `import` statement like `DEFAULT_WAIT_TIME` and use it in this while loop? that would be better for code maintenance.

Reply:
c6dda6e